### PR TITLE
Arun

### PR DESCRIPTION
**Pull request by aaronpxavier**
Created signup form.

### CODE REVIEW

File: src/app/utility/validators/passwordMismatch.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import {AbstractControl, ValidationErrors} from '@angular/forms';
+
+export const passwordMismatch = (abstractControl: AbstractControl): void => {
+  const password = abstractControl.get('password');
+  const confirmPassword = abstractControl.get('confirmPassword');
+
+  if (password.value && password.value && password.value === confirmPassword.value) {
```
Comment (dereklance) [inline src/app/utility/validators/passwordMismatch.ts]: checking `password.value` twice

**Review by dereklance — CHANGES_REQUESTED** [review]

**Review by dereklance — APPROVED** [review]

File: src/app/components/signup/signup.component.ts [hunk 2]
```diff
@@ -1,5 +1,16 @@
-import {AfterContentInit, Component, OnInit} from '@angular/core';
+import {AfterContentInit, Component, OnInit, ViewChild} from '@angular/core';
```
Comment (ronhenson) [inline src/app/components/signup/signup.component.ts:1]: remove Viewchild - not used

File: src/app/components/signup/signup.component.ts [hunk 3]
```diff
@@ -8,12 +19,86 @@ import {ToolbarService} from '../../services/utility/toolbar.service';
...
+  public isPending = false;
+  public isError = false;
+  public showForm = true;
+  public confirmationMsg = '';
+  public showConfirmationMsg = false;
+  private readonly token: string;
+
+  constructor(
+    private toolbarService: ToolbarService,
+    private authService: AuthService,
```
Comment (ronhenson) [inline src/app/components/signup/signup.component.ts:35]: refactor change name authservice to userService

File: src/app/services/api/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+
+export class AuthService {
+  URL = environment.FLIGHTS_API;
+
+  constructor(private http: HttpClient) {}
+
+  postUser(payload: any): any {
```
Comment (ronhenson) [inline src/app/services/api/auth.service.ts:14]: change "any" type by using interface

Comment (ronhenson) [inline src/app/services/api/auth.service.ts:14]: insert try catch

File: src/app/services/api/auth.service.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    const url = this.URL + '/auth/sign-up';
```
Comment (ronhenson) [inline src/app/services/api/auth.service.ts:15]: techdebt: naming the api auth to user??

File: src/app/utility/validators/passwordMismatch.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import {AbstractControl, ValidationErrors} from '@angular/forms';
+
+export const passwordMismatch = (abstractControl: AbstractControl): void => {
+  const password = abstractControl.get('password');
+  const confirmPassword = abstractControl.get('confirmPassword');
+
+  if (password.value && password.value === confirmPassword.value) {
```
Comment (ronhenson) [inline src/app/utility/validators/passwordMismatch.ts:7]: parenthesis to make easier to read password.value confirmPassword

**Pull request merged**